addon/authenticators/oauth2-implicit-grant.js

```diff
@@ -0,0 +1,118 @@
+import Ember from 'ember';
+import BaseAuthenticator from './base';
+
+const {
+  RSVP,
+  isEmpty
+} = Ember;
+
+/**
+ Authenticator that conforms to OAuth 2
+ ([RFC 6749](http://tools.ietf.org/html/rfc6749)), specifically the _"Implicit
+ Grant Type"_.
+
+ @class OAuth2ImplicitGrantAuthenticator
+ @module ember-simple-auth/authenticators/oauth2-implicit-grant
+ @extends BaseAuthenticator
+ @public
+ */
+export default BaseAuthenticator.extend({
+  /**
+   Restores the session from a session data object; __will return a resolving
+   promise when there is a non-empty `access_token` in the session data__ and
+   a rejecting promise otherwise.
+
+   @method restore
+   @param {Object} data The data to restore the session from
+   @return {Ember.RSVP.Promise} A promise that when it resolves results in the session becoming or remaining authenticated
+   @public
+   */
+  restore(data) {
+    return new RSVP.Promise((resolve, reject) => {
+      if (!this._validateToken(data)) {
+        return reject();
+      }
+
+      return resolve(data);
+    });
+  },
+
+  /**
+   Authenticates the session using the specified `response`
+   (see https://tools.ietf.org/html/rfc6749?hmm=lol#section-4.2.2); if no response
+   ist set the response will be parsed from `window.location.hash`.
+
+   __If the access token is valid and thus authentication succeeds, a promise that
+   resolves with the access token is returned__, otherwise a promise that rejects
+   with the error as returned by the server is returned
+   (see https://tools.ietf.org/html/rfc6749?hmm=lol#section-4.2.2.1).
+
+   @method authenticate
+   @param {Object} response The response
+   @return {Ember.RSVP.Promise} A promise that when it resolves results in the session becoming authenticated
+   @public
+   */
+  authenticate(response) {
+    return new RSVP.Promise((resolve, reject) => {
+      if (isEmpty(response)) {
+        response = this._parseResponse(window.location.hash);
+      }
+
+      if (response.error) {
+        reject(response);
+      } else if (!this._validateToken(response)) {
+        reject({
+          error: 'invalid_token',
+          // jscs:disable requireCamelCaseOrUpperCaseIdentifiers
+          error_description: 'The received token is invalid'
+          // jscs:enable requireCamelCaseOrUpperCaseIdentifiers
+        });
+      } else {
+        resolve(response);
+      }
+    });
+  },
+
+  /**
+   This method simply returns a resolving promise.
+
+   @method invalidate
+   @return {Ember.RSVP.Promise} A promise that when it resolves results in the session being invalidated
+   @public
+   */
+  invalidate() {
+    return RSVP.Promise.resolve();
+  },
+
+  _validateToken(token) {
```

Comment:
As this validates a data set and not just the token I think `_validateData(data)` might be a better name?